pyrobosim/pyrobosim/core/robot.py

```diff
@@ -421,6 +421,20 @@ def navigate(
 
         return self.follow_path(path, realtime_factor=realtime_factor)
 
+    def reset_path_planner(self):
+        """Resets the robot's path planner, if available."""
+        if self.path_planner is None:
+            warnings.warn(f"[{self.name}] Robot has no path planner. Cannot reset.")
+            return
+
+        if not hasattr(self.path_planner, "reset"):
```

Comment:
@eholum this was inspired by your question on the `reset()` method being required. Now it's technically not